prelude/bigint.md

```diff
@@ -18,36 +18,36 @@ impl BigInt {
 
     /// Computes `self / other`, returns None if `other` is zero.
     pub fn checked_div(other: impl Into<BigInt>) -> Option<BigInt>;
+}
 
-    /// Returns the unique value that is equal to `self` modulo `2^size.bits()`.
-    /// If `signed == Unsigned`, the result is in the interval `0..2^size.bits()`,
-    /// else it is in the interval `-2^(size.bits()-1) .. 2^(size.bits()-1)`.
-    ///
-    /// `size` must not be zero.
-    pub fn modulo(self, signed: Signedness, size: Size) -> BigInt {
-        if size == 0 {
-            panic!("BigInt::modulo received invalid size zero!");
-        }
-
-        // the modulus.
-        let m = BigInt::from(2).pow(size.bits());
-
-        // n is in range `-(m-1)..m`.
-        let n = self % m;
-
-        match signed {
-            // if `Unsigned`, output needs to be in range `0..m`:
-            Unsigned if n < 0 => n + m,
-            // if `Signed`, output needs to be in range `-m/2 .. m/2`:
-            Signed if n >= m/2 => n - m,
-            Signed if n < -m/2 => n + m,
-            _ => n,
-        }
+/// Returns the unique value that is equal to `i` modulo `2^size.bits()`.
+/// If `signed == Unsigned`, the result is in the interval `0..2^size.bits()`,
+/// else it is in the interval `-2^(size.bits()-1) .. 2^(size.bits()-1)`.
+///
+/// `size` must not be zero.
+pub fn modulo(i: BigInt, signed: Signedness, size: Size) -> BigInt {
```

Comment:
My proposal was to move these functions to specr entirely. Here in Minirust we just want something like
````
For documentation purposes, some of the key BigInt functions are listed here, but they are all defined by specr.

```rust,ignore
impl BigInt {
    /// Returns the unique value that is equal to `i` modulo `2^size.bits()`.
    /// If `signed == Unsigned`, the result is in the interval `0..2^size.bits()`,
    /// else it is in the interval `-2^(size.bits()-1) .. 2^(size.bits()-1)`.
    ///
    /// `size` must not be zero.
    pub fn modulo(i: BigInt, signed: Signedness, size: Size) -> BigInt
}
```
````

Reply:
I think we cannot define it in specr, if we want specr to be a crate.
modulo uses `Size` and `Signedness` after all, right?

Reply:
Right, so those types would have to move to specrlib. Hm...

In Coq we have `Z.modulo` which already always returns a positive number, not like the strange `%` operation Rust has. But we'd still have to duplicate signedness...